pkg/test/masteruserrecord/masteruserrecord_assertion.go

```diff
@@ -201,50 +219,144 @@ func (a *Assertion) userAccountHasTier(ua toolchainv1alpha1.UserAccountEmbedded,
+	assert.Equal(a.t, value, a.mur.Labels[key])
 	return a
 }
 
 // HasAnnotationWithValue verifies that the MasterUserRecord has
 // an annotation with the given key and value
-func (a *Assertion) HasAnnotationWithValue(key, value string) *Assertion {
-	err := a.loadUaAssertion()
+func (a *MasterUserRecordAssertion) HasAnnotationWithValue(key, value string) *MasterUserRecordAssertion {
+	err := a.loadMasterUserRecord()
+	require.NoError(a.t, err)
+	require.Contains(a.t, a.mur.Annotations, key)
+	assert.Equal(a.t, value, a.mur.Annotations[key])
+	return a
+}
+
+func (a *MasterUserRecordAssertion) HasOriginalSub(sub string) *MasterUserRecordAssertion {
+	err := a.loadMasterUserRecord()
+	require.NoError(a.t, err)
+	assert.Equal(a.t, sub, a.mur.Spec.OriginalSub)
+	return a
+}
+
+func (a *MasterUserRecordAssertion) HasTargetCluster(targetcluster string) *MasterUserRecordAssertion {
+	err := a.loadMasterUserRecord()
+	require.NoError(a.t, err)
+	require.NotEmpty(a.t, a.mur.Spec.UserAccounts)
+	assert.Equal(a.t, targetcluster, a.mur.Spec.UserAccounts[0].TargetCluster)
+	return a
+}
+
+func (a *MasterUserRecordAssertion) HasUserAccounts(count int) *MasterUserRecordAssertion {
+	err := a.loadMasterUserRecord()
+	require.NoError(a.t, err)
+	require.Len(a.t, a.mur.Spec.UserAccounts, count)
+	return a
+}
+
+func (a *MasterUserRecordAssertion) HasUserAccountTierName(tiername string) *MasterUserRecordAssertion {
+	err := a.loadMasterUserRecord()
+	require.NoError(a.t, err)
+	assert.Equal(a.t, tiername, a.mur.Spec.UserAccounts[0].Spec.NSTemplateSet.TierName)
+	return a
+}
+
+func (a *MasterUserRecordAssertion) HasUserAccountNamespaceTemplateRefs(refs ...string) *MasterUserRecordAssertion {
+	err := a.loadMasterUserRecord()
+	require.NoError(a.t, err)
+	nsRefs := make([]toolchainv1alpha1.NSTemplateSetNamespace, len(refs))
+	for i, ref := range refs {
+		nsRefs[i] = toolchainv1alpha1.NSTemplateSetNamespace{
+			TemplateRef: ref,
+		}
+	}
+	assert.Equal(a.t, nsRefs, a.mur.Spec.UserAccounts[0].Spec.NSTemplateSet.Namespaces)
+	return a
+}
+
+func (a *MasterUserRecordAssertion) HasUserAccountClusterResourceTemplateRefs(ref string) *MasterUserRecordAssertion {
+	err := a.loadMasterUserRecord()
+	require.NoError(a.t, err)
+	assert.Equal(a.t, &toolchainv1alpha1.NSTemplateSetClusterResources{
+		TemplateRef: ref,
+	}, a.mur.Spec.UserAccounts[0].Spec.NSTemplateSet.ClusterResources)
+	return a
+}
+
+// Assertions on multiple MasterUserRecords at once
+
+type MasterUserRecordsAssertion struct {
+	murs      *toolchainv1alpha1.MasterUserRecordList
+	client    client.Client
+	namespace string
+	t         test.T
+}
+
+func AssertThatMasterUserRecords(t test.T, namespace string, client client.Client) *MasterUserRecordsAssertion {
+	return &MasterUserRecordsAssertion{
+		client:    client,
+		namespace: namespace,
+		t:         t,
+	}
+}
+
+func (a *MasterUserRecordsAssertion) loadMasterUserRecords() error {
+	if a.client != nil {
```

Comment:
why do we need this check? The client should be always set, right?

Reply:
right, I'm not sure why I added that. Probably a quick copy/paste from https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/test/config/toolchainconfig_assertion.go#L22, but I will remove it

Reply:
removed (both) in d93f97b